### PR TITLE
Added dev requirements

### PR DESCRIPTION
**Pull request by raddessi**
This issue was raised in https://github.com/digitalocean/pynetbox/issues/329

### CODE REVIEW

**Comment by raddessi** [comment]
It looks like the tests here are failing again due to the issues fixed in https://github.com/digitalocean/pynetbox/pull/327. I'm not sure the best way to show a dependency in github without additional apps, do you have any recommendations for that @markkuleinio or @zachmoody ? Rebasing does not seem to be the solution from my quick testing.

File: requirements-dev.txt
Comment (raddessi) [inline requirements-dev.txt:5]: Technically pyyaml is pulled in via some of these other packages it's not a terrible idea to list it anyway.

**Comment by markkuleinio** [comment]
> I'm not sure the best way to show a dependency in github without additional apps, do you have any recommendations for that @markkuleinio or @zachmoody ?

Unfortunately I don't have a clue what's happening

**Comment by raddessi** [comment]
Sorry about that :/ Can I try to clear anything up?

**Comment by markkuleinio** [comment]
> I'm sorry about that :/ Can I try to clear anything up?

I'd suggest you make a new branch from pynetbox master, and try to push some small PR with it, and make sure that the process goes through cleanly. That would ensure other contributors can push PRs as well.

**Comment by raddessi** [comment]
I don't 100% follow.. isn't that what's being done here?

**Comment by raddessi** [comment]
Oh I see you mean set up a testing branch. If I thought to do it that would have been a good ask, but when zach had merged the initial integration testing PR it was passing tests just fine and had been for a while. The reason they are broken is due to a mistake I made in not pinning a repo version to use for the testing of the most recent netbox version, which I do have a PR in to fix but zach seems to be busy. I apologize on that, a testing branch would have been a good idea but I didn't think about it at the time and I guess neither did Zach. For now, I have PRs up to fix any known issues and get tests stable again.

**Comment by markkuleinio** [comment]
I did this:
- synced upstream pynetbox to my fork
- created new branch "fix-pickling"
- committed my changes
- created fresh venv, with "black==19.10b0 pytest pytest-docker faker" (as mentioned in the py3.yml workflow)
- ran pytest, which gave me lots of errors, like

/usr/lib/python3.7/subprocess.py:1522: FileNotFoundError
===================================== short test summary info ======================================
ERROR tests/integration/test_dcim.py::TestSimpleServerRackingAndConnecting::test_racking_server[netbox v2.7]
ERROR tests/integration/test_pytest_docker.py::test_netbox_version[netbox v2.7] - FileNotFoundErr...
ERROR tests/integration/test_dcim.py::TestSimpleServerRackingAndConnecting::test_racking_server[netbox v2.8]
ERROR tests/integration/test_pytest_docker.py::test_netbox_version[netbox v2.8] - FileNotFoundErr...
ERROR tests/integration/test_dcim.py::TestSimpleServerRackingAndConnecting::test_racking_server[netbox v2.9]
ERROR tests/integration/test_pytest_docker.py::test_netbox_version[netbox v2.9] - FileNotFoundErr...
ERROR tests/integration/test_dcim.py::TestSimpleServerRackingAndConnecting::test_racking_server[netbox v2.10]
ERROR tests/integration/test_pytest_docker.py::test_netbox_version[netbox v2.10] - FileNotFoundEr...
================================== 286 passed, 8 errors in 6.34s ===================================

- pushed my changes anyway
- created a PR https://github.com/digitalocean/pynetbox/pull/331
- saw that the checks failed in pynetbox repo (unrelated to my changes as far as I see it).

So my questions are:
1. What do I have to change to get my own pytest working again when developing pynetbox?
1. What does someone else have to change so that PR checks are successful again? If I understood you correctly you are waiting for @zachmoody to accept some fixes that you have already done for this.

Thanks for your testing code efforts, in any case!

**Comment by raddessi** [comment]
OK interesting, I definitely get errors when I try to reproduce this locally but they are not `FileNotFound` errors, they are the errors from the upstream testing failures like you had mentioned.

To your questions:
> What do I have to change to get my own pytest working again when developing pynetbox?

Up until now all I had known was needed was PR https://github.com/digitalocean/pynetbox/pull/327 to me merges so that upstream and local integration tests would run without error again. I'm not sure what this issue you have is though so there may be something else causing issues. Can you run unit tests only without error right now via `pytest tests/test_* tests/unit/*`? If so then those are the only ones that would possibly work correctly until #327 gets merged due to upstream changes in netbox-docker.

Once #327 goes in all that should be required is to run `pytest` after installing the packages like you have already.. I would really like to see what your output is when you run #327 locally.. 

> What does someone else have to change so that PR checks are successful again? If I understood you correctly you are waiting for @zachmoody to accept some fixes that you have already done for this.

Yeah, I apologize I think this was bad timing. Very soon after the first integration testing was merged netbox-docker made some changes that broke our tests, only because I had missed one spot to pin a specific repo version. So that is what #327 is all about, once that gets merged in theory all tests will run for anyone with the required packages installed. It may be different in a venv.. but I can't off the top of my head think up a good reason as to why. I've used conda envs for this thing plenty of times. Anyway, I think we are just waiting on a few PRs to be merged in.. like I said I'm sorry, this one is on me :/

Are you on the NetworkToCode slack? I didn't see you on a quick check. If you want to do some quick troubleshooting some time soon my name should be the same there as well. If not.. can you post some more detail about what file went missing in the test?

**Comment by markkuleinio** [comment]
Sorry for the delay, this is the script:
- create venv
- install the same packages as in the github workflow (black==19.10b0 pytest pytest-docker faker)
- clone pynetbox
- move to pynetbox directory
- run pytest

And here are the full results: https://pastebin.com/behzbE1b

File: .github/workflows/py3.yml
Comment (zachmoody) [inline .github/workflows/py3.yml:29]: I'm cool with just installing pynetbox under dev requirements

```suggestion
      - name: Install dev requirements
        run: pip install -r requirements-dev.txt .
```

Should do it.

File: requirements-dev.txt
Comment (zachmoody) [inline requirements-dev.txt:4]: We'll probably want to pin these so we don't run into the problem we saw in #327 (if I'm understanding that one correctly).

```suggestion
Faker==6.5.*
pytest==6.2.*
pytest-docker==0.10.*
```

**Review by zachmoody — CHANGES_REQUESTED** [review]
Looks good, just a couple tweaks.

File: requirements-dev.txt
Comment (raddessi) [reply requirements-dev.txt:4]: Oh yeah, good call there also. Thanks!

**Comment by raddessi** [comment]
I updated with the recommended changes, but I want to test out the issue mark reported in https://github.com/digitalocean/pynetbox/pull/330#issuecomment-788010015 just to see if I can verify locally now before this goes in.

File: .github/workflows/py3.yml
Comment (raddessi) [inline .github/workflows/py3.yml:26]: Nice.. I thought requirements files and individual module names had to be ran via separate commands which is why I had separated these initially but this works just fine. Thanks for the pointer

Comment (zachmoody) [reply .github/workflows/py3.yml:26]: Yep, thought the same thing. Gave it a shot though and it seemed to work. :man_shrugging:

**Pull request merged**
